### ISSUE TITLE
`which` not working on windows

### ISSUE THREAD

**Issue by maximousblk** [ddanier/nur#20]
**Describe the bug**
The `which` (nu built-in) commnad doesnt work on the windows version, while it does on the linux version.

**To Reproduce**
1. Install latest `nur@0.5.1` 
2. define a task
   ```sh
   def "nur w" [] { which which }
   ```
3. Run the `nur w` task
4. Notice that on windows the command fails with `nu::shell::external_command`

![image](https://github.com/ddanier/nur/assets/28438021/681f5850-a0de-4ca7-9a30-872b97ac2969)

**Expected behavior**
The `which` command runs successfully and prints the following output

![image](https://github.com/ddanier/nur/assets/28438021/664060c6-4e30-498a-a5c9-958dd8cd5c9c)

**Version details (please complete the following information):**
 - OS: windows 
 - OS-Version: 11 (22631)
 - nur Version: 0.5.1

**Additional context**
N/A

**Comment by ddanier** [comment]
Thanks for the report! I will look into this.

It seems like I have to look at the registration of all commands again....there might be something different happening on Windows.

**Comment by ddanier** [comment]
Background is that nu does only adds `which` when the feature flag `which-support` is used. This also means that `which` on Linux is not actually working as of now - but Linux systems provide a `which` cli version. I will enable the `which-support` by default as it makes no actual sense to skip this command + this is the nu default anyways.

For comparison on Mac I have the following results running `which which` with 0.5.1 and dev version with this feature flag active:
```
❯ nur -c "which which"
nur version 0.5.1
Project path: /Users/ddanier/work/nur
Running command: which which

/usr/bin/which
Task execution successful

❯ cargo run -- -c "which which"
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.72s
     Running `target/debug/nur -c 'which which'`
nur version 0.5.1
Project path: /Users/ddanier/work/nur
Running command: which which

╭───┬─────────┬──────┬──────────╮
│ # │ command │ path │   type   │
├───┼─────────┼──────┼──────────┤
│ 0 │ which   │      │ built-in │
╰───┴─────────┴──────┴──────────╯

Task execution successful
```

You see the difference here, it will just call the cli `which` in 0.5.1 and use the `which` builtin command in the dev version.

**Comment by ddanier** [comment]
Fixed with version `0.5.2` 😉
